bin/varnishd/cache/cache_req_fsm.c

```diff
@@ -78,6 +80,68 @@
 REQ_STEPS
 #undef REQ_STEP
 
+/*--------------------------------------------------------------------
+ * Remember additional objcores to deref when the task ends
+ */
+
+struct ocstash {
+	unsigned		magic;
+#define OCSTASH_MAGIC		0xe361b34d
+	unsigned		malloced;
+	unsigned		n;
+	unsigned		l;
+	struct objcore *	ocs[] v_counted_by_(l);
+};
+
+static void
+ocstash_fini(struct worker *wrk, struct ocstash **stashp)
+{
+	struct ocstash *stash;
+	unsigned u;
+
+	AN(stashp);
+	if (*stashp == NULL)
+		return;
+	TAKE_OBJ_NOTNULL(stash, stashp, OCSTASH_MAGIC);
+	assert(stash->n <= stash->l);
+	for (u = 0; u < stash->n; u++)
+		(void)HSH_DerefObjCore(wrk, &stash->ocs[u], HSH_RUSH_POLICY);
```

Comment:
We can't take any objcore and assume they're all supposed to wake `rush_exponent` requests up.

Reply:
Right. The call in `cnt_fetch` which changed the rush parameter from 1 to HSH_RUSH_POLICY is actually not needed - we only need to consider the cases where VCL had a chance to look at the object.

Reply:
But you made me realize that I had overlooked `sub vcl_hit {}` in `cnt_lookup()`, which has access to `obj.*`. Thank you!

Reply:
I should have said "after `vcl_hash`" in my review comment, because a grace hit could actually take significant time to resume.

Reply:
A grace hit is immediate. Do you mean a waitinglist hit?

Reply:
A grace hit needs to wait for the request to be released, which can take significant time for a couple reasons. First, the bgfetch task may be queued if we are out of threads. Second, if the request has a body, it can take until it's actually consumed, at which point at least one timeout is involved (again, orders of magnitude above milliseconds).